main.tf

```
@@ -69,3 +69,47 @@ resource "ibm_resource_tag" "key_protect_tag" {
   tags        = var.access_tags
   tag_type    = "access"
 }
+
+##############################################################################
+# Context Based Restrictions
+##############################################################################
+
+locals {
+  default_operations = [{
+    api_types = [
+      {
+        "api_type_id" : "crn:v1:bluemix:public:context-based-restrictions::::api-type:"
+      },
+      {
+        "api_type_id" : "crn:v1:bluemix:public:context-based-restrictions::::platform-api-type:"
+      }
+    ]
+  }]
+}
+
+module "cbr_rule" {
+  count            = length(var.cbr_rules) > 0 ? length(var.cbr_rules) : 0
+  source           = "terraform-ibm-modules/cbr/ibm//modules/cbr-rule-module"
+  version          = "1.23.0"
+  rule_description = var.cbr_rules[count.index].description
+  enforcement_mode = var.cbr_rules[count.index].enforcement_mode
+  rule_contexts    = var.cbr_rules[count.index].rule_contexts
+  resources = [{
+    attributes = [
+      {
+        name  = "accountId"
+        value = var.cbr_rules[count.index].account_id
+      },
+      {
+        name     = "serviceInstance"
+        value    = ibm_resource_instance.key_protect_instance.guid
+        operator = "stringEquals"
+      },
+      {
+        name  = "serviceName"
+        value = "kms"
+      }
+    ]
+    operations = var.cbr_rules[count.index].operations == null ? local.default_operations : var.cbr_rules[count.index].operations
+  }]
+}
```

Operations value will come after  the `resources` for the condition to pick both platform and service API types  

```suggestion
    ]
  }]
   operations = var.cbr_rules[count.index].operations == null ? local.default_operations : var.cbr_rules[count.index].operations
}
```